src/cargo/ops/lockfile.rs

```diff
@@ -247,6 +248,14 @@ fn emit_package(dep: &toml::Table, out: &mut String) {
 }
 
 fn lock_root(ws: &Workspace<'_>) -> Filesystem {
+    if let Some(requested) = ws.requested_lockfile_path() {
```

Comment:
If we had `Workspace::lockfile_path()`, this function can be rewritten to

```rust
fn lock_root(...) {
    if ws.root_maybe().is_embedded() {
        ws.target_dir()
    } else {
        ws.lockfile_path().parent().expect("lockfile path exists")
    }
}
```
What do you think?

Reply:
Yep that's what I was originally thinking of doing too actually! Let me fix that

Reply:
Moved `lock_root()` all together into `worspace`, to keep current behavior. It also looks like the idea of `Filesystem` is to hold directory path to work with files inside it.
LMK if that's what you were thinking 
132ba16779eb2c28d43796a82160ce02348158d1